platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,896 @@
+/**
```

Comment:
I should probably add here that my current refactoring plan for this class is to split apart all the logic into classes that can represent each of the possible bindings. So all the different ways of doing business for redirect will be in a redirect binding object that knows how to do all the parsing and validation for that scenario. Then I can just simply use the correct binding object for the operation I'm trying to do. That should make this class pretty small. If you have any different ideas for how I might be able to better refactor this, lemme know!